web-common/src/features/sources/sourceUtils.ts

```diff
@@ -37,6 +41,25 @@ export function compileCreateSourceYAML(
   return `${topLineComment}\n\ntype: "${connectorName}"\n` + compiledKeyValues;
 }
 
+function buildDuckDbQuery(path: string): string {
+  const extension = extractFileExtension(path as string);
+  if (containsAny(extension, [".csv", ".tsv", ".txt"])) {
+    return `select * from read_csv_auto('${path}')`;
+  } else if (containsAny(extension, [".parquet"])) {
+    return `select * from read_parquet('${path}')`;
+  } else if (containsAny(extension, [".json", ".ndjson"])) {
+    return `select * from read_json('${path}',auto_detect=true,format='auto')`;
+  }
+  return `select * from '${path}'`;
+}
+
+function containsAny(fullExtensions: string, extensions: Array<string>) {
```

Comment:
I don't follow what "fullExtensions" means. It looks like it's just an "extension"

Reply:
This stems from the go APIs to get extension. It only extracts the last part.